org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/references/utils/XMLReferencesUtils.java

```diff
@@ -150,102 +155,168 @@ private static void searchToAttributes(DOMAttr fromAttr, List<XMLReferenceExpres
+	private static void collectToNode(DOMNode fromNode, XMLReferencesSearcher referenceSearcher,
+			boolean matchNode, IXMLReferenceTosCollector collector, DOMElement toElement,
 			String toNamespacePrefix) {
-		if (toElement.hasAttributes()) {
-			NamedNodeMap toAttributes = toElement.getAttributes();
-			if (toAttributes != null) {
-				for (int j = 0; j < toAttributes.getLength(); j++) {
-					DOMAttr toAttr = (DOMAttr) toAttributes.item(j);
-					XMLReferenceExpression expression = findExpressionWhichMatchesTo(fromAttr, toAttr,
-							referenceExpressions,
-							matchAttr);
-					if (expression != null) {
-						collector.collect(toNamespacePrefix, toAttr, expression);
+		if (referenceSearcher.isSearchInAttribute()) {
+			// Search to reference in attribute nodes
+			if (toElement.hasAttributes()) {
+				NamedNodeMap toAttributes = toElement.getAttributes();
+				if (toAttributes != null) {
+					for (int j = 0; j < toAttributes.getLength(); j++) {
+						DOMAttr toAttr = (DOMAttr) toAttributes.item(j);
+						XMLReferenceExpression expression = findExpressionWhichMatchesTo(fromNode, toAttr,
+								referenceSearcher,
+								matchNode);
+						if (expression != null) {
+							collector.collect(toNamespacePrefix, toAttr, expression);
+						}
 					}
 				}
 			}
 		}
+		if (referenceSearcher.isSearchInText()) {
+			// Search to reference in text node.
+			DOMNode firstChild = toElement.getFirstChild();
+			if (firstChild != null && firstChild.isText()) {
+				XMLReferenceExpression expression = findExpressionWhichMatchesTo(fromNode, firstChild,
+						referenceSearcher,
+						matchNode);
+				if (expression != null) {
+					collector.collect(toNamespacePrefix, firstChild, expression);
+				}
+			}
+		}
 	}
 
 	/**
-	 * Returns the reference expression where the given attributes fromAttr and
-	 * toAttr matches an expression from the given referenceExpressions.
+	 * Returns the reference expression where the given attributes
+	 * <code>fromNode</code> and
+	 * <code>toNode</code> matches an expression from the given
+	 * referenceSearcher.
 	 * 
-	 * @param fromAttr             the from attribute.
-	 * @param toAttr               the to attribute.
-	 * @param referenceExpressions reference expressions which matches the from
-	 *                             attribute.
-	 * @param matchAttr            true if the test of the value of from and to
-	 *                             attribute must be checked and false otherwise.
-	 * @return the reference expression where the given attributes fromAttr and
-	 *         toAttr matches an expression from the given referenceExpressions.
+	 * @param fromNode          the from attribute, text node.
+	 * @param toNode            the to attribute, text node.
+	 * @param referenceSearcher reference searcher which matches the from
+	 *                          node.
+	 * @param matchNode         true if the test of the value of from and to

```

Comment:
true if the value of the from and to nodes must be checked, and false otherwise

Reply:
fixed